sql/core/src/main/scala/org/apache/spark/sql/execution/command/CommandUtils.scala

```diff
@@ -43,13 +43,17 @@ object CommandUtils extends Logging {
   /** Change statistics after changing data by commands. */
   def updateTableStats(sparkSession: SparkSession, table: CatalogTable): Unit = {
     val catalog = sparkSession.sessionState.catalog
-    if (sparkSession.sessionState.conf.autoSizeUpdateEnabled) {
-      val newTable = catalog.getTableMetadata(table.identifier)
-      val newSize = CommandUtils.calculateTotalSize(sparkSession, newTable)
-      val newStats = CatalogStatistics(sizeInBytes = newSize)
-      catalog.alterTableStats(table.identifier, Some(newStats))
-    } else if (table.stats.nonEmpty) {
-      catalog.alterTableStats(table.identifier, None)
+    val statsMaybeUpdatedByHive = catalog.getTableMetadata(table.identifier)
+    if (statsMaybeUpdatedByHive.stats == table.stats) {
+      if (sparkSession.sessionState.conf.autoSizeUpdateEnabled) {
+        val newSize = CommandUtils.calculateTotalSize(sparkSession, statsMaybeUpdatedByHive)
+        val newStats = CatalogStatistics(sizeInBytes = newSize)
+        catalog.alterTableStats(table.identifier, Some(newStats))
+      } else if (table.stats.nonEmpty) {
+        catalog.alterTableStats(table.identifier, None)
+      }
+    } else {
+      logInfo(s"Skip to update statistics for ${table.qualifiedName} as it updated by Hive.")
```

Comment:
The difference of the statistics doesn't mean the correctness of the stats, does it?